guides/keras_cv/classification_with_keras_cv.py

```diff
@@ -0,0 +1,688 @@
+
+image_batch = next(iter(eval_ds.take(1)))["images"]
+keras_cv.visualization.plot_image_gallery(
+    image_batch,
+    rows=3,
+    cols=3,
+    value_range=(0, 255),
+    show=True,
+)
+keras_cv.visualization.plot_image_gallery(
+    image_batch,
+    rows=3,
+    cols=3,
+    value_range=(0, 255),
+    show=True,
+)
+
+"""
+Finally, lets unpackage our datasets and prepare to pass them to the `model.fit()`
+call, which accepts a tuple of `(images, labels)`.
+"""
+
+
+def unpackage_dict(inputs):
+    return inputs["images"], inputs["labels"]
+
+
+train_ds = train_ds.map(unpackage_dict, num_parallel_calls=tf.data.AUTOTUNE)
+eval_ds = eval_ds.map(unpackage_dict, num_parallel_calls=tf.data.AUTOTUNE)
+
+"""
+Cool!  Data augmentation is by far the hardest piece of training a classifier
+in the modern era.
+Congratulations on making it this far!
+
+## Optimizer Tuning
+
+To achieve optimal performance, we must implement a Warm up Cosinde decay
+learning rate schedule.
+While we won't go into detail on this schedule, [you can read more about it
+here](https://scorrea92.medium.com/cosine-learning-rate-decay-e8b50aa455b).
+"""
+
+
+def lr_warmup_cosine_decay(
+    global_step,
+    warmup_steps,
+    hold=0,
+    total_steps=0,
+    start_lr=0.0,
+    target_lr=1e-2,
+):
+    # Cosine decay
+    learning_rate = (
+        0.5
+        * target_lr
+        * (
+            1
+            + tf.cos(
+                tf.constant(math.pi)
+                * tf.cast(global_step - warmup_steps - hold, tf.float32)
+                / float(total_steps - warmup_steps - hold)
+            )
+        )
+    )
+
+    warmup_lr = tf.cast(target_lr * (global_step / warmup_steps), tf.float32)
+    target_lr = tf.cast(target_lr, tf.float32)
+
+    if hold > 0:
+        learning_rate = tf.where(
+            global_step > warmup_steps + hold, learning_rate, target_lr
+        )
+
+    learning_rate = tf.where(global_step < warmup_steps, warmup_lr, learning_rate)
+    return learning_rate
+
+
+class WarmUpCosineDecay(schedules.LearningRateSchedule):
+    def __init__(self, warmup_steps, total_steps, hold, start_lr=0.0, target_lr=1e-2):
+        super().__init__()
+        self.start_lr = start_lr
+        self.target_lr = target_lr
+        self.warmup_steps = warmup_steps
+        self.total_steps = total_steps
+        self.hold = hold
+
+    def __call__(self, step):
+        lr = lr_warmup_cosine_decay(
+            global_step=step,
+            total_steps=self.total_steps,
+            warmup_steps=self.warmup_steps,
+            start_lr=self.start_lr,
+            target_lr=self.target_lr,
+            hold=self.hold,
+        )
+
+        return tf.where(step > self.total_steps, 0.0, lr, name="learning_rate")
+
+
+"""
+Next let's construct this optimizer:
+"""
+
+total_steps = (9000 // BATCH_SIZE) * EPOCHS
+warmup_steps = int(0.1 * total_steps)
+hold_steps = int(0.45 * total_steps)
+schedule = WarmUpCosineDecay(
```

Comment:
Any chance we could plot this bad boy?

Reply:
I plotted it out and got https://i.imgur.com/YCr5pII.png

I couldn't think of a good way to do it in the nodebook because we have to overwrite EPOCHS and all of the variables - so it would add a lot of noise